hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -1123,9 +1177,13 @@ private void rollbackInflightCommits() {
     }
   }
 
-  private HoodieTable getTableAndInitCtx(JavaRDD<HoodieRecord<T>> records) {
+  private HoodieTable getTableAndInitCtx(String operationType) {
+    HoodieTableMetaClient metaClient = createMetaClient(true);
+    if (operationType.equalsIgnoreCase(DELETE_STR)) {
```

Comment:
Can you move the call addSchemaToConfig(metaClient) to delete() as it should only be called from there ?
Also, rename addSchemaToConfig to setWriteSchemaFromLastInstant ?

Reply:
I thought about it, but it needs MetaClient. For every other operation we create the metaClient within getTableAndInitCtx. addSchemaToConfig also needs metaClient and hence I placed it here. Let me know how you want to go about. we could create Metaclient upfront, but all operations will have to create it and send it to getTableAndInitCtx() as an arg.

Reply:
Based on our offline conversation, leaving it as is.